presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -164,10 +180,119 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         }
     }
 
+    public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint constraint)
+    {
+        requireNonNull(session, "session is null");
```

Comment:
No need to validate input arguments to methods. If null is passed, the caller will get a direct NPE.

Reply:
Removed.